pg_spot_operator/cloud_impl/aws_vm.py

```diff
@@ -388,9 +388,9 @@ def ec2_launch_instance(
     network_interface: dict[str, Any] = {
         "AssociatePublicIpAddress": False,
         "DeviceIndex": 0,
-        "DeleteOnTermination": False,
+        "DeleteOnTermination": m.floating_ips,
```

Comment:
Do you mean `not m.floating_ips` ?

Reply:
Actually correct as-is, floating = delete on termination and get a new IP next time